src/ioc-react/useDependencies.ts

```diff
@@ -0,0 +1,9 @@
+import { useContext } from 'react';
+import { DependencyContainerReactContext } from '../ioc-react';
+import { mapDependenciesToProps } from './mapDependenciesToProps';
+
+export function useDependencies<T>(diKeys: T) {
+  const { container } = useContext(DependencyContainerReactContext);
+
+  return mapDependenciesToProps<{ [propName in keyof T]: any }>(container, diKeys);
```

Comment:
lets soak this in...
```
const () => {

    const { a } = useDependencies({ a: Ass })

    const [a] = useDependencies(Ass)

    const [] = useState()

    const [a, b] = useDependencies(Ass, Bat)
}
```

Reply:
Ryan and jon prefer array syntax, so...

- Let's have useDependencies returning an array
- Let's add useDependency returning the singular thing